Library/Homebrew/dev-cmd/tests.rb

```diff
@@ -37,7 +37,7 @@ def tests
 
       Homebrew.install_gem_setup_path! "bundler"
       unless quiet_system("bundle", "check")
-        system "bundle", "install", "--path", "vendor/bundle"
+        system "bundle", "install"
```

Comment:
It's not the default, but there is likely a `.bundle/config` in the same directory as the `Gemfile` that specifies this behavior. It is created the first time you run `bundle install` with `--path`.